### PR TITLE
[X86][MC] Add assembly alias for CCMP, CTEST, SETZUCC, CMPCCXADD

### PR DESCRIPTION
**Pull request by FreddyLeaf**

### CODE REVIEW

File: llvm/lib/Target/X86/X86InstrAsmAlias.td [hunk 1]
```diff
@@ -124,19 +124,31 @@ def : InstAlias<"ccmp"#Cond#"{q} $dcf\t{$src2, $src1|$src1, $src2}",
 defm : CCMP_Aliases<"o" ,  0>;
 defm : CCMP_Aliases<"no",  1>;
 defm : CCMP_Aliases<"b" ,  2>;
+defm : CCMP_Aliases<"c" ,  2>;
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrAsmAlias.td:127]: What does `c` mean?

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86InstrAsmAlias.td:127]: `carry`

Comment (KanRobert) [reply llvm/lib/Target/X86/X86InstrAsmAlias.td:127]: https://www.godbolt.org/z/G1nv15xar

Comment (KanRobert) [reply llvm/lib/Target/X86/X86InstrAsmAlias.td:127]: Why do you want to add it?

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86InstrAsmAlias.td:127]: First, binutil on godbolt is too old. binutil has updated to 2.43, while it only has 2.38

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86InstrAsmAlias.td:127]: My objective is to align with gas.

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86InstrAsmAlias.td:127]: binutils reference: https://github.com/bminor/binutils-gdb/blob/b935898fd1b323a47452e672de597a497bdeadf6/opcodes/i386-opc.tbl#L350

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**